benchmarks/layeredflow/layeredflow.cc

```diff
@@ -0,0 +1,461 @@
+                            const double epsilon)
+      {
+        const double y0 = 1./3.;
+        const double yplus=(1.+y0)/beta;
+        const double yminus=(1.-y0)/beta;
+        const double C1 = 2.*numbers::PI /
+                          ( beta*std::log(beta*beta+pow(1.+y0,2.0))-2.*(1.+y0)*std::atan(yplus)
+                            - beta*std::log(beta*beta+pow(1.-y0,2.0))+2.*(1.-y0)*std::atan(yminus)
+                            + 2.*numbers::PI*(1.+2.*epsilon) );
+
+        const double C2 = ( beta*std::log( beta*beta+std::pow(1+y0,2.) )- 2.*(1.+y0)*std::atan((1.+y0)/beta)
+                            + numbers::PI*(1.+2.*epsilon) )*C1 ;
+        const double y = pos[1]-1.0;
+        const double v_x = (-beta*C1*std::log(std::pow(beta,2.0)+std::pow(y-y0,2.0))+2.*(y-y0)*C1*std::atan((y-y0)/beta)
+                            + numbers::PI*(1.+2.*epsilon)*y*C1+C2)/(2.*numbers::PI);
+        const double v_y = 0;
+        return Point<2> (v_x,v_y);
+      }
+
+      double
+      LayeredFlow_pressure (const Point<2> &,
+                            const double beta,
+                            const double epsilon)
+      {
+        return 0;
+      }
+
+
+      template <int dim>
+      class FunctionLayeredFlow : public Function<dim>
+      {
+        public:
+          FunctionLayeredFlow (const double beta, const double epsilon)
+            :
+            Function<dim>(dim+2),
+            beta_(beta),
+            epsilon_(epsilon)
+          {}
+
+          virtual void vector_value (const Point< dim >   &pos,
+                                     Vector< double >   &values) const
+          {
+            Assert (dim == 2, ExcNotImplemented());
+            Assert (values.size() >= 4, ExcInternalError());
+
+            const Point<2> p (pos[0], pos[1]);
+
+            const Tensor<1,2> v = AnalyticSolutions::LayeredFlow_velocity (p, beta_, epsilon_);
+            values[0] = v[0];
+            values[1] = v[1];
+            values[2] = AnalyticSolutions::LayeredFlow_pressure (p, beta_, epsilon_);
+          }
+
+        private:
+          const double beta_;
+          const double epsilon_;
+      };
+    }
+
+
+
+    template <int dim>
+    class LayeredFlowBoundary : public BoundaryVelocity::Interface<dim>, public aspect::SimulatorAccess<dim>
+    {
+      public:
+        /**
+         * Constructor.
+         */
+        LayeredFlowBoundary();
+
+        /**
+         * Return the boundary velocity as a function of position.
+         */
+        virtual
+        Tensor<1,dim>
+        boundary_velocity (const types::boundary_id ,
+                           const Point<dim> &position) const;
+
+      private:
+        const double beta;
+        const double epsilon;
+    };
+
+
+    /**
+     * @note This benchmark only talks about the flow field, not about a
+     * temperature field. All quantities related to the temperature are
+     * therefore set to zero in the implementation of this class.
+     _r*
```

Comment:
not wrong to put the `_r` here, but probably neither intended to supposed to convey meaning :-)

Reply:
no idea where this came from :)